nixos/modules/system/boot/systemd.nix

```diff
@@ -451,6 +451,21 @@ in
         cfg.services
     );
 
+    assertions = concatLists (
+      mapAttrsToList
+        (name: service:
+          map (message: {
+            assertion = false;
+            inherit message;
+          }) (concatLists [
+            (optional ((builtins.elem "network-interfaces.target" service.after) || (builtins.elem "network-interfaces.target" service.wants))
```

Comment:
I think this is a fine set of things to check. It could be mentioned in a bunch of other places, such as conflicts or before, and likely others.

Frankly this assertion, though requested, feels like gold-plating. If it helps this review go through, fine.

Reply:
The most common usages were in those parameters. I didn't find a sane way to just check all the places where it could occur.